src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java

```diff
@@ -761,6 +761,28 @@ void deleteRecordedQueue(String queue) {
             this.maybeDeleteRecordedAutoDeleteExchange(b.getSource());
         }
     }
+    
+    /**
+     * Remove the queue from list of recorded queues to recover after connection failure.
+     * Intended to be used in usecases where you want to remove the queue from this connection's recovery list but don't want to delete the queue from the server.
+     * 
+     * @param queue queue name to remove from recorded recovery queues
+     * @param ifUnused if true, the RecordedQueue will only be removed if no more consumers from this connection are using it.
+     */
+    public void removeRecordedQueue(final String queue, final boolean ifUnused) {
```

Comment:
WDYT about `excludeQueueFromRecovery` or `doNotRecoveryQueue`?

Reply:
I like doNotRecoverQueue or removeQueueFromRecovery.
excludeQueueFromRecovery works too. I think we would just need to make it clear you have to call the method after declaring a queue and not before. Its not adding an exclude filter or anything like that for future declareQueue calls.

Reply:
My personal favorite is `excludeQueueFromRecovery`.

Reply:
I'd go with `excludeQueueFromRecovery`.

Reply:
works for me :)
I'll change that and look at adding a junit.
Thanks guys.